soroban-sdk/src/logs.rs

```diff
@@ -143,7 +149,7 @@ impl testutils::Logs for Logs {
         let log_sym = ScSymbol(StringM::try_from("log").unwrap());
         let log_topics = ScVec(VecM::try_from(vec![ScVal::Symbol(log_sym)]).unwrap());
         env.host()
-            .get_events()
+            .get_diagnostic_events()
```

Comment:
Was the all function broken and getting the wrong events?

Reply:
This error is in CI:
```
error[E0599]: no method named `get_diagnostic_events` found for reference `&soroban_env_host::Host` in the current scope
   --> soroban-sdk/src/logs.rs:152:14
    |
151 | /         env.host()
152 | |             .get_diagnostic_events()
    | |             -^^^^^^^^^^^^^^^^^^^^^ help: there is a method with a similar name: `set_diagnostic_level`
    | |_____________|
    | 
```

Reply:
The all function externalizes non diagnsotic events, which affects the budget. Also the error is happening because this PR depends on https://github.com/stellar/rs-soroban-env/pull/1411, but it doesn't reference it yet. I tested it with a local patch.